### PR TITLE
[llvm-driver] Add driver alias for c++filt

### PR DESCRIPTION
**Pull request by rupprecht**
Similar to 9d955a63c7b0d2d547a63d4074e03cd5ebd1f162, this allows the driver to respond to `c++filt`.

### CODE REVIEW

File: utils/bazel/llvm-project-overlay/llvm/driver.bzl [hunk 1]
```diff
@@ -39,8 +39,10 @@ _EXTRA_ALIASES = {
     "clang": ["clang++", "clang-cl", "clang-cpp"],
     "lld": ["ld", "lld-link", "ld.lld", "ld64.lld", "wasm-ld"],
     "llvm-ar": ["ranlib", "lib", "dlltool"],
+    "llvm-cxxfilt": ["c++filt"],
     "llvm-objcopy": ["bitcode-strip", "install-name-tool", "strip"],
     "llvm-objdump": ["otool"],
+    "llvm-profdata": ["gcov-tool"],
```
[the remark this answers is not in the document]
Comment (rupprecht) [reply utils/bazel/llvm-project-overlay/llvm/driver.bzl:45]: It doesn't, but this alias allows one to build `llvm` and then symlink `gcov-tool` to it to have it work as llvm-profdata.

[the remark this answers is not in the document]
Comment (MaskRay) [reply utils/bazel/llvm-project-overlay/llvm/driver.bzl:45]: While `llvm-cov gcov` emulates `gcov`, llvm-profdata is an entirely different tool and doesn't emulate `gcov-tool`.

gcov-tool and llvm-profdata are so different. If llvm-profdata symlinked to gcov-tool works, it indicates that gcov-tool is just needed by the build system but isn't actually used.

[the remark this answers is not in the document]
Comment (rupprecht) [reply utils/bazel/llvm-project-overlay/llvm/driver.bzl]: I'm not entirely convinced, but the problem is small enough in scope that I can update callers to call if `llvm-profdata` instead. I reverted this part, now the patch is just handling `c++filt`.

> gcov-tool and llvm-profdata are so different. If llvm-profdata symlinked to gcov-tool works, it indicates that gcov-tool is just needed by the build system but isn't actually used.

It seems like even if they generally aren't compatible, the most popular use case is `$tool merge -o ...`, for which both `llvm-profdata merge -o ...` and `gcov-tool merge -o ...` work, so it ends up appearing compatible.

**Review by MaskRay — APPROVED** [review]

**Comment by MaskRay** [comment]
FTR: gcov-tool might be used for GCC PGO/AutoFDO. The tool is different from llvm-profdata even if they both support `xxx merge xxx yyy -o zzz`

`gcc -fprofile-generate -c a.c b.c && gcc -fprofile-generate a.o b.o -o a; ./a; gcov-tool merge . . -o dir`; gcov-tool creates a directory.

`clang -fprofile-generate -c a.c b.c && fclang -fprofile-generate a.o b.o -o a; llvm-profdata merge dir1 dir2 -o file`; llvm-profdata creates a file.

**Pull request merged**